### ISSUE TITLE
Bug: [`perfectionist/sort-intersection-types`] indentation can impact rule interpretation

### ISSUE THREAD

**Issue by OlivierZal** [azat-io/eslint-plugin-perfectionist#181]
### Describe the bug

In the screenvideo below, you can see 2 strange behaviours:
- the rule works but is broken when changing the indentation
- when inverting 2 types as requested by the error message, a new error message is raised and contradicts the previous one

### Code example

Project (for testing – remove the `eslint-disable` comment): https://github.com/OlivierZal/melcloud-api/blob/fd854ab272e3b818b0570878d9815e938481d7ec/src/facades/interfaces.ts#L61

```
setHolidayMode: (({
    enabled,
    from,
    days,
  }: {
    enabled?: true
    from?: string
    days: number
  }) => Promise<FailureData | SuccessData>) &
    (({
      enabled,
      from,
      to,
    }: {
      enabled?: true
      from?: string
      to: string
    }) => Promise<FailureData | SuccessData>) &
    (({ enabled }: { enabled: false }) => Promise<FailureData | SuccessData>)
```

### ESLint version

v9.7.0

### ESLint Plugin Perfectionist version

v3.0.0

### Additional comments

Here is a screenvideo:

https://github.com/user-attachments/assets/b8c1cb15-589a-493e-a52f-2f846e006b1f

### Validations

- [X] Read the [docs](https://perfectionist.dev).
- [X] Check that there isn't [already an issue](https://github.com/azat-io/eslint-plugin-perfectionist/issues) that reports the same bug to avoid creating a duplicate.

**Comment by azat-io** [comment]
Hello, I have tried to reproduce your problem. And everything seems to be ok and nothing strange happens when deleting comments that disable ESLint rule.

Are you sure it's not Prettier that changes the indentation size?

**Comment by OlivierZal** [comment]
You're right, the indentation change is due to prettier, but 2 comments:
- in my opinion the rule should not care of stylistically / indentation, only of sorting (which should not depend on how it is indented)
- even if stylistics / indentation is taken into account, you can see in the screenshot that the perfectionist error is not consistent since it changes each time we invert 2 signatures according to the raised error (once the error requires "days" to be first, and when "days" is first it requires "to" to be first): the error should be consistent over the time, and not fall into an endless loop

**Comment by OlivierZal** [comment]
You're right, the indentation change is due to prettier.

However:
- in my opinion the rule should not care of stylistically / indentation, only of sorting (which should not depend on how it is indented)

- even if stylistics / indentation is taken into account, you can see in the following of the video that the perfectionist error is not consistent since it changes each time we invert 2 signatures according to the raised error (the error requires "days" to be first, and when "days" is put first, then it requires "to" to be first, etc.): the error should be consistent over the time, and not fall into an endless loop

**Comment by OlivierZal** [comment]
An argument for why non-stylistic ESLint rules shouldn’t enforce stylistics is that other rules, such as `sort-switch-case`, do not either.

e.g. the following won't raise any error despite the wrong indent:

```
  #getTargetTemperatureRange(operationMode = this.setData.OperationMode): {
    max: number
    min: number
  } {
    switch (operationMode) {
      case OperationMode.auto:
        return {
          max: this.data.MaxTempAutomatic,
          min: this.data.MinTempAutomatic,
        }
                  case OperationMode.cool: // wrongly indented
      case OperationMode.dry:
        return { max: this.data.MaxTempCoolDry, min: this.data.MinTempCoolDry }
      case OperationMode.heat:
      default:
        return { max: this.data.MaxTempHeat, min: this.data.MinTempHeat }
    }
  }
  ```

**Comment by hugop95** [comment]
Hi @OlivierZal ! I would like to look deeper into this issue, is it still a problem? I can't reproduce it locally.

**Comment by OlivierZal** [comment]
Hi @OlivierZal, thanks for having tried. I confirm the bug still exists. Do you a Slack or Discord or whatever which could ease some explanations about what happens?

**Comment by OlivierZal** [comment]
Hi @hugop95, thanks for having tried :-)

I confirm the bug still exists. Do you a Slack or Discord or whatever which could ease some explanations about what happens?

**Comment by OlivierZal** [comment]
Basically, to reproduce:
- fork the project below
- remove the `eslint-disable`
- apply eslint --> fixes an issue
- add a space before the `enabled` which was under the `eslint-disable` --> an error appears, although indentation should not impact

**Comment by hugop95** [comment]
@OlivierZal Unfortunately, after cloning the project and testing on `main`, I still can't seem to reproduce the issue. Maybe I'm doing something wrong? We can talk in more details about it on Slack indeed: https://join.slack.com/shareDM/zt-2rdmtqq7k-10xoybXJmHbYj7wS4H9xMw .

**Comment by hugop95** [comment]
Alright @OlivierZal , I have identified the problem, take the following example that is identical to your issue:

```ts
type T =
{ a: string } &
{ b: string }
``` 
This code is correct, the first node's name is `{ a: string}` and the second `{ b: string }}`.

This code however (notice the additional space before `b`
```ts
type T =
{ a: string} &
{  b: string}
```
is not, because ` ` has a lower unicode value than any alphanumeric character, so `{ a` > `{  b`.

One solution is to ignore spaces altogether. This theoretically impacts all rules, I will study the subject a bit and see if this solution could cause issues.
